timesync/ticker.go

```diff
@@ -0,0 +1,180 @@
+package timesync
+
+import (
+	"errors"
+	"github.com/spacemeshos/go-spacemesh/common/types"
+	"github.com/spacemeshos/go-spacemesh/log"
+	"sync"
+	"time"
+)
+
+// subs implements a lock-protected Subscribe-Unsubscribe structure
+// note: to access internal fields a lock must be obtained
+type subs struct {
```

Comment:
you can use it in manual clock as well as real clock

Reply:
Yes, but since you can't use the lock (private) I preferred to duplicate.
They are not really connected. It is ok.
I would even rather avoid this type of map "LayerTimer". Don't find any reason for it but decided to keep the PR as simple as possible considering our objectives.